src/main/java/com/iota/iri/utils/IotaUtils.java

```diff
@@ -101,4 +109,61 @@ public static ExecutorService createNamedSingleThreadExecutor(String name) {
         return Executors.newSingleThreadExecutor(r -> new Thread(r, name));
 
     }
+	
+	/**
+     * Parses a human readable string for a file size (tb, gb, mb -> file modifier)
+     * Follows the following format: [double][optional space][case insensitive file modifier]
+     * 
+     * Kb is not parsed as this is too small for us to be used in a persistence provider
+     * 
+     * @return The size of the human readable string in bytes, rounded down.
+     */
+    public static long parseFileSize(String humanReadableSize) {
+        humanReadableSize = humanReadableSize.replaceAll(",", "").toLowerCase();
+        int spaceNdx = humanReadableSize.indexOf(" ");
+        double ret;
+        //If we forgot a space,check until we find it
+        if (spaceNdx == -1) {
+            spaceNdx = 0;
+            while (spaceNdx < humanReadableSize.length() && 
+                    (Character.isDigit(humanReadableSize.charAt(spaceNdx)) || humanReadableSize.charAt(spaceNdx) == '.' )) {
+                spaceNdx++;
+            }
+
+            // Still nothing? started with a character
+            if (spaceNdx == 0) {
+                return -1;
+            }
+            ret = Double.parseDouble(humanReadableSize.substring(0, spaceNdx));
+        } else {
+            // ++ to skip the space
+            ret = Double.parseDouble(humanReadableSize.substring(0, spaceNdx++));
+        }
+
+        //Default to GB
+        String sub = ret == humanReadableSize.length() ? "GB" : humanReadableSize.substring(spaceNdx);
+        switch (sub) {
+            case "tb":
+                return (long) (ret * TB_FACTOR);
+            case "tib":
+                return (long) (ret * TIB_FACTOR);
+            case "gb":
+                return (long) (ret * GB_FACTOR);
+            case "gib":
+                return (long) (ret * GIB_FACTOR);
+            case "mb":
+                return (long) (ret * MB_FACTOR);
+            case "mib":
+                return (long) (ret * MIB_FACTOR);
```

Comment:
the entire thing is meanth for usability, not because we want to support mb in db size pruning

Reply:
But currently user can put MB no?

Reply:
> hmmm no this seems faulty as spacendx already is -1 itll neverbe a .

Didn't understand, but if you found a bug fix it